docs/rfcs/001-updateable-bundled-packages.md

```diff
@@ -0,0 +1,100 @@
+
+> One paragraph explanation of the feature.
+
+This feature will enable an opt-in subset of bundled Atom packages to be updated with `apm` outside of the Atom release cycle.  This will enable users to receive new functionality and bug fixes for some bundled packages as regularly as needed without waiting for them to be included in a new Atom release.  This is especially important for packages like [GitHub](https://github.com/atom/github/) and  [Teletype](https://github.com/atom/teletype/) which provide essential Atom functionality and could be improved independently of Atom.
+
+## Motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+
+Atom currently uses a monthly release cycle with staged Stable and Beta releases so that major issues get caught early in Beta before reaching the Stable release.  Because Atom releases updates monthly, this means that a new feature merged into `master` right after a new Atom release could take one month to reach the next Beta and then another month to reach Stable.
+
+Since a large part of Atom's built-in functionality is provided by bundled packages, it makes sense to allow some of those packages to be updated independently of Atom's monthly release cycle so that users can receive new features and fixes whenever they become available.
+
+The primary use case for this improvement is enabling the GitHub package to ship improvements more frequently than Atom's release cycle since many of its improvements can be done without changes to Atom itself.  If this approach is proven to work well for the GitHub package, we might also consider using it to ship Teletype as a bundled Atom package.
+
+## Explanation
+
+> Explain the proposal as if it was already implemented in Atom and you were describing it to an Atom user. That generally means:
+> - Introducing new named concepts.
+> - Explaining the feature largely in terms of examples.
+> - Explaining any changes to existing workflows.
+
+Bundled packages are treated differently than community packages that you can install using `apm`:
+
+- You are not prompted to update them when new versions are released on `apm`
+- `apm` will warn you at the command line when you try to install or update a bundled package
+- If a user intentionally installs a bundled package from `apm` the [Dalek package](https://github.com/atom/dalek/) will show a warning in the "deprecations" view asking the user to remove the offending package
+
+Despite all this, if the user *does* install a bundled package using `apm`, it will be loaded into the editor and updated dutifully as releases occur.
+
+### Implementation Details
+
+Because the necessary infrastructure is already in place to enable updates to bundled packages using `apm`, the only work required is to provide a way for packages to opt in to this behavior and for `apm` to include those packages in its update checks if they haven't already been installed in the user's packages folder.
```

Comment:
> ... if they haven't already been installed in the user's packages folder.

I don't understand the relevance of this clause. (Sorry. 😬) Shouldn't `apm` always include these packages when checking for updates?

Reply:
I'll tighten up the language here so that it's clearer!

Basically what I was trying to convey is that `apm` will check bundled packages for updates *right now* if you've installed one manually with `apm` and it's in your `~/.atom/packages` folder, but it needs to be taught how to find bundled packages to update when they haven't been installed using `apm` yet.